### PR TITLE
Initial development pull reuqest.

### PR DESCRIPTION
**Pull request by oxied**
http://redmine.weezlabs.com:8100/redmine/boards/8/topics/56
http://redmine.weezlabs.com:8100/redmine/boards/8/topics/57

### CODE REVIEW

File: app/src/main/java/com/example/admin/lab_56_57/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Button for launching activity for Lab #56
+        Button launcher56 = (Button) findViewById(R.id.launcher56);
+
+        launcher56.setOnClickListener(new View.OnClickListener() {
+                                          @Override
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/MainActivity.java]: Fix indentations. https://source.android.com/source/code-style.html. Paragraph "Use Spaces for Indentation"

File: app/src/main/java/com/example/admin/lab_56_57/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+public class MainActivity extends AppCompatActivity {
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Button for launching activity for Lab #56
+        Button launcher56 = (Button) findViewById(R.id.launcher56);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/MainActivity.java]: Rename to `launchActivity56Button`

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.gradle
```
Comment (oxied) [inline .gitignore]: Seems gitignore was rewrited by old version. Please, change to the github version for android.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package com.example.admin.lab_56_57;
+
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+
+
+public class ActivityForLab57 extends AppCompatActivity {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java:9]: No need to rename now, but it's better to use this name: Lab57Activity. In this way it matches naming of MainActivity, for instance.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_activity_for_lab57);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java]: double activity_activity. remove one of them.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+public class ActivityForLab57Fragment extends android.support.v4.app.Fragment {
+
+    private AdapterForLinear adapterForLinear_;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: The same thing. We're used to use naming rule: from private to general. So class should be named as LinearAdapter and field as linearAdapter_.
no need to change it right now. But keep it in mind.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    private AdapterForRelative adapterForRelative_;
+    private Spinner menuSpinner_;
+    private ArrayAdapter<String> spinnerAdapter_;
+    private final String LINEAR = "Linear Layout";
+    private final String RELATIVE = "Relative Layout";
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Use enum intead of these two constants.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_activity_for_lab57, container, false);
+
+        // Get values for spinner
+        String[] valuesForSpinner = getResources().getStringArray(R.array.spinner);
+
+        // List of objects for title and description
+        List<AppInfo> values = new ArrayList<AppInfo>();
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Not the best name for parameter. Change to appInfoValues

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+        PackageManager packageManager = getActivity().getPackageManager();
+        List<ApplicationInfo> packages = packageManager.getInstalledApplications(0);
+
+
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Unnecessary empty line.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+            double size = Utility.getSize(packageManager,packageInfo);
+            // Getting icon of app
+            Drawable icon = Utility.getIcon(packageManager,packageInfo);
+            // Adding all values to object
+            AppInfo singleItem = new AppInfo(packageInfo.packageName, packageInfo.targetSdkVersion, size,icon);
+            values.add(singleItem);
+        }
+
+
+
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Remove this and above empty lines.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        listView.setAdapter(adapterForLinear_);
+
+        spinnerAdapter_ = new ArrayAdapter<String>(
+                getActivity(),
+                R.layout.support_simple_spinner_dropdown_item,
+                valuesForSpinner
+        );
+
+
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Unnecessary line.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        menuSpinner_ = (Spinner) rootView.findViewById(R.id.menu_spinner);
+        menuSpinner_.setAdapter(spinnerAdapter_);
+
+        menuSpinner_.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
+                String city = parent.getItemAtPosition(pos).toString();
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: What does the city means?

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+                } else if (city.equals(RELATIVE)) {
+                    listView.setAdapter(adapterForRelative_);
+                    adapterForLinear_.notifyDataSetChanged();
+                }
+
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+                // TODO Auto-generated method stub
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Remove TODO comment if you are not going to TODO something in the future. For instance, rename to `// Nothing to do`

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * Custom adapter for big list
+ */
+public class AdapterForBigList extends ArrayAdapter<ItemInfo> {
+
+    /**
+     * Represent the subitems of menuItem like TextView etc. for ViewHolder patern
+     */
+    static class ViewHolder {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Make it private if it's not used anywhere outside of the wrapper class.

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        TextView nameOfItem;
+        TextView descriptionOfItem;
+        ImageView image;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Rename these 3 fields as we discusses.

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public AdapterForBigList(Context context, List<ItemInfo> values) {
+        super(context, R.layout.big_list_item, values);
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Unnecessary empty space.

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * Sets the image depends on the position
+     *
+     * @param viewHolder
+     * @param imageId
+     */
+    private void setImage(ViewHolder viewHolder, int imageId) {
+
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Unnecessary empty space.

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForLinear.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+public class AdapterForLinear extends ArrayAdapter<AppInfo> {
+
+    public AdapterForLinear(Context context, List<AppInfo> values){
+        super(context, R.layout.linear_list_item, values);
+
+    }
+
+    @Override
+    public View getView(int position, View convertView,ViewGroup parent){
+        View view = LayoutInflater.from(getContext()).inflate(R.layout.linear_list_item, parent, false);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForLinear.java]: Forgot to use ViewHolder pattern here

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForRelative.java [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+ */
+public class AdapterForRelative extends ArrayAdapter<AppInfo> {
+
+    public AdapterForRelative(Context context, List<AppInfo> values) {
+        super(context, R.layout.relative_list_item, values);
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        View view = LayoutInflater.from(getContext()).inflate(R.layout.relative_list_item, parent, false);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForRelative.java]: Forgot to use ViewHolder pattern here

File: app/src/main/java/com/example/admin/lab_56_57/BigListActivityFragment.java [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+            values.add(singleItem);
+            fictionalId ++;
+            if (fictionalId >4) {
+                fictionalId = 1;
+            }
+
+        }
+
+        ListView listView = (ListView) rootView.findViewById(R.id.big_list_view);
+        adapterForBigList_ = new AdapterForBigList(this.getActivity(),values);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/BigListActivityFragment.java]: apply formatting to this part: this.getActivity(),values

File: app/src/main/java/com/example/admin/lab_56_57/MainActivity.java [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+                                          }
+                                      }
+
+        );
+
+        // Button for launching activity for Lab #57
+        Button launcher57 = (Button) findViewById(R.id.launcher57);
+
+        launcher57.setOnClickListener(new View.OnClickListener() {
+                                          @Override
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/MainActivity.java]: The same.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+import android.content.pm.PackageManager;
+import android.graphics.drawable.Drawable;
+import android.util.Log;
+
+import java.io.File;
+
+/**
+ * Utility class
+ */
+public class Utility {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: It's better to name it Utils.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+        } else return false;
+    }
+
+    /**
+     * Creates description for app
+     * @param version
+     * @param size
+     * @return
+     */
+    public static String makeDescription(int version,double size){
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: Apply formatting to this line.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+        return description;
+    }
+
+    /**
+     * Get icon
+     * @param packageManager
+     * @param packageInfo
+     * @return
+     */
+    public static Drawable getIcon(PackageManager packageManager,ApplicationInfo packageInfo){
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: Apply formatting to this line.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+        return icon;
+    }
+
+    /**
+     * Get size of app
+     * @param packageManager
+     * @param packageInfo
+     * @return
+     */
+    public static double getSize (PackageManager packageManager,ApplicationInfo packageInfo){
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: Apply formatting to this line.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+    public static Drawable getIcon(PackageManager packageManager,ApplicationInfo packageInfo){
+        // Getting icon of app
+        Drawable icon = packageInfo.loadIcon(packageManager);
+        return icon;
+    }
+
+    /**
+     * Get size of app
+     * @param packageManager
+     * @param packageInfo
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: Remove unnecessary lines like param without description. leave only significant javadoc fields.

File: app/src/main/java/com/example/admin/lab_56_57/Utility.java [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+     * @return
+     */
+    public static double getSize (PackageManager packageManager,ApplicationInfo packageInfo){
+        double size = 0;
+        try {
+            File file = new File(packageInfo.sourceDir);
+            size = file.length();
+        } catch (Exception e) {
+            Log.e("ERROR", e.getMessage());
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/Utility.java]: If you use TAG, it's better to declare it in one single point (Application class, Utils class, for instance).

File: app/src/main/java/com/example/admin/lab_56_57/data/AppInfo.java [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+
+    public double getSize() {
+        return size;
+    }
+
+    public Drawable getIcon() {
+        return icon;
+    }
+
+    public AppInfo(String appname, int version, double size, Drawable icon) {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/AppInfo.java]: Put constructor in the beginning of class right after fields declaration.

File: app/src/main/java/com/example/admin/lab_56_57/data/ItemInfo.java [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+    public int getImageId() {
+        return imageId;
+    }
+
+    public ItemInfo(String title, String description) {
+        this.title = title;
+        this.description = description;
+    }
+
+    public ItemInfo(int imageId, String title, String description) {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/ItemInfo.java]: The same.

File: app/src/main/java/com/example/admin/lab_56_57/data/AppInfo.java [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+/**
+ * Describes different information about app
+ */
+public class AppInfo {
+    private String title;
+    private int targetSdkVersion;
+    private double size;
+    private Drawable icon;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/AppInfo.java]: private fields should end with `_`

File: app/src/main/java/com/example/admin/lab_56_57/data/ItemInfo.java [hunk 31]
```diff
@@ -0,0 +1,37 @@
...
+ * Created by Mikhail Valuyskiy on 21.05.2015.
+ */
+
+/**
+ * Represent the item in the ListView
+ */
+public class ItemInfo {
+    private String title;
+    private String description;
+    private int imageId;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/ItemInfo.java]: private fields should end with `_`

File: app/src/main/res/layout/activity_main.xml [hunk 32]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical" >
+
+    <Button
+        android:id="@+id/launcher56"
```
Comment (oxied) [inline app/src/main/res/layout/activity_main.xml]: rename to launch_56_activity_button.

File: app/src/main/res/layout/activity_main.xml [hunk 33]
```diff
@@ -0,0 +1,20 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Launch Lab #56" />
+
+    <Button
+        android:id="@+id/launcher57"
```
Comment (oxied) [inline app/src/main/res/layout/activity_main.xml]: The same.

File: app/src/main/res/layout/activity_main.xml [hunk 34]
```diff
@@ -0,0 +1,20 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical" >
+
+    <Button
+        android:id="@+id/launcher56"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Launch Lab #56" />
```
Comment (oxied) [inline app/src/main/res/layout/activity_main.xml]: Use resources.

File: app/src/main/res/layout/activity_main.xml [hunk 35]
```diff
@@ -0,0 +1,20 @@
...
+
+    <Button
+        android:id="@+id/launcher57"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Launch Lab #57" />
```
Comment (oxied) [inline app/src/main/res/layout/activity_main.xml]: The same.

File: app/src/main/res/layout/big_list_item.xml [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+    <ImageView
+        android:id="@+id/big_list_item_image_view"
+        android:src="@mipmap/ic_launcher"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+    <LinearLayout
+        android:layout_weight="1"
+        android:orientation="vertical"
+        android:paddingLeft="30dp"
```
Comment (oxied) [inline app/src/main/res/layout/big_list_item.xml]: Use resources for paddings.

File: app/src/main/res/layout/big_list_item.xml [hunk 37]
```diff
@@ -0,0 +1,36 @@
...
+    <TextView
+        android:id="@+id/big_list_item_title"
+        android:text="@string/title"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+    <TextView
+        android:id="@+id/big_list_item_details"
+        android:text="@string/description"
+        android:paddingTop="10dp"
```
Comment (oxied) [inline app/src/main/res/layout/big_list_item.xml]: Use resources for paddings.

File: app/src/main/res/layout/fragment_activity_for_lab57.xml [hunk 38]
```diff
@@ -0,0 +1,24 @@
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (oxied) [inline app/src/main/res/layout/fragment_activity_for_lab57.xml]: Rename file to lab57_activity_fragment.

File: app/src/main/res/layout/linear_list_item.xml [hunk 39]
```diff
@@ -0,0 +1,49 @@
...
+    <ImageView
+        android:id="@+id/list_item_image_view"
+        android:src="@mipmap/ic_launcher"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+    <LinearLayout
+        android:layout_weight="1"
+        android:orientation="vertical"
+        android:paddingLeft="30dp"
```
Comment (oxied) [inline app/src/main/res/layout/linear_list_item.xml]: Use resources for paddings.

File: app/src/main/res/layout/linear_list_item.xml [hunk 40]
```diff
@@ -0,0 +1,49 @@
...
+        <TextView
+            android:id="@+id/list_item_title"
+            android:text="@string/title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content" />
+
+        <TextView
+            android:id="@+id/list_item_details"
+            android:text="@string/description"
+            android:paddingTop="10dp"
```
Comment (oxied) [inline app/src/main/res/layout/linear_list_item.xml]: Use resources for paddings.

File: app/src/main/res/layout/linear_list_item.xml [hunk 41]
```diff
@@ -0,0 +1,49 @@
...
+    <LinearLayout
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:orientation="vertical">
+
+        <ImageView
+            android:id="@+id/list_item_small_image"
+            android:src="@mipmap/ic_launcher"
+            android:layout_width="25dp"
+            android:layout_height="25dp" />
```
Comment (oxied) [inline app/src/main/res/layout/linear_list_item.xml]: Use resources for dimensions.

File: app/src/main/res/layout/relative_list_item.xml [hunk 42]
```diff
@@ -0,0 +1,38 @@
...
+        android:id="@+id/list_item_image_view"
+        android:src="@mipmap/ic_launcher"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+
+    <TextView
+        android:id="@+id/list_item_title"
+        android:text="@string/title"
+        android:paddingLeft="30dp"
```
Comment (oxied) [inline app/src/main/res/layout/relative_list_item.xml]: Use resources for dimensions.

File: app/src/main/res/layout/relative_list_item.xml [hunk 43]
```diff
@@ -0,0 +1,38 @@
...
+        android:layout_toRightOf="@+id/list_item_image_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+    <TextView
+        android:id="@+id/list_item_details"
+        android:text="@string/description"
+        android:paddingTop="10dp"
+        android:paddingLeft="30dp"
```
Comment (oxied) [inline app/src/main/res/layout/relative_list_item.xml]: Use resources for dimensions.

File: app/src/main/res/layout/relative_list_item.xml [hunk 44]
```diff
@@ -0,0 +1,38 @@
...
+        android:layout_toRightOf="@+id/list_item_image_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+
+    <ImageView
+        android:id="@+id/list_item_small_image"
+        android:src="@mipmap/ic_launcher"
+        android:layout_alignParentRight="true"
+        android:layout_width="25dp"
+        android:layout_height="25dp" />
```
Comment (oxied) [inline app/src/main/res/layout/relative_list_item.xml]: Use resources for dimensions.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java (hunk 12 above)
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: What does the city mean?

File: app/src/main/res/values/array.xml [hunk 45]
```diff
@@ -0,0 +1,38 @@
...
+        <item>The capital of Thailand</item>
+        <item>The capital of England</item>
+    </string-array>
+
+    <string-array name="spinner">
+        <item>Linear Layout</item>
+        <item>Relative Layout</item>
+    </string-array>
+
+</resources>
```
Comment (oxied) [inline app/src/main/res/values/array.xml:7]: Is this file necessary?

File: app/src/main/res/values/strings.xml [hunk 46]
```diff
@@ -0,0 +1,14 @@
...
+    <string name="app_name">Lab_56_57</string>
+
+    <!--All values for list -->
+    <string name="title">Title</string>
+    <string name="description">Description</string>
+
+    <!--All values for action bar menu -->
+    <string name="action_settings">Settings</string>
+    <string name = "linear_layout">Linear Layout</string>
+    <string name = "lab">Next lab</string>
```
Comment (oxied) [inline app/src/main/res/values/strings.xml]: Wrong formatting.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java (hunk 5 above)
Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java]: Changed name for layout of activity

File: app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java [hunk 47]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * Custom adapter for big list
+ */
+public class AdapterForBigList extends ArrayAdapter<ItemInfo> {
+
+    /**
+     * Represent the subitems of menuItem like TextView etc. for ViewHolder pattern
+     */
+    private static class ViewHolder_ {
+        TextView nameOfItemView;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Should add underscore not to the class name but to its fields.

Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/AdapterForBigList.java]: Also it's better to name this field like `itemNameView` or just `nameTextView`. I'd prefer the last one.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 48]
```diff
@@ -0,0 +1,116 @@
...
+ * A placeholder fragment containing a simple view.
+ */
+public class ActivityForLab57Fragment extends android.support.v4.app.Fragment {
+
+    private AdapterForLinear adapterForLinear_;
+    private AdapterForRelative adapterForRelative_;
+    private Spinner menuSpinner_;
+    private ArrayAdapter<String> spinnerAdapter_;
+
+    private enum Mode_ {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Mode without underscore

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 49]
```diff
@@ -0,0 +1,116 @@
...
+        LINEAR("Linear Layout"),
+        RELATIVE("Relative Layout");
+
+        private final String text_;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: For now I don't see that you use this field, but I see how you can use it. Think and let me know your thoughts.

Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: For using enum with String it is necessary to create a private member variable and a private constructor must be defined. That is why I created this field

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java (hunk 4 above)
Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57.java:9]: Will try with future projects

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java (hunk 49 above)
Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Fixed

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 50]
```diff
@@ -0,0 +1,140 @@
...
+            AppInfo singleItem = new AppInfo(packageInfo.packageName, packageInfo.targetSdkVersion, size,icon);
+            appInfoValues.add(singleItem);
+        }
+
+        ListView listView = (ListView) rootView.findViewById(R.id.list_view);
+
+        relativeAdapter_ = new AdapterForRelative(this.getActivity(), appInfoValues);
+        linearAdapter_ = new AdapterForLinear(this.getActivity(), appInfoValues);
+
+        listView.setAdapter(linearAdapter_);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: As you understand, this line hides a potential problem: if you make a mistake in menuSpinner's onItemSelected, activity will still work and the user will not get a chance to understand that there is no difference in each of modes cause modes just don't work. Guess to leave adapter setting only in one place in onItemSelected.

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 51]
```diff
@@ -0,0 +1,140 @@
...
+        });
+
+        return rootView;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        ListView listView = (ListView) getActivity().findViewById(R.id.list_view);
+        listView.setAdapter(relativeAdapter_);
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Is it good idea to store the mode in SPreferences and restore already chosen one?

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 52]
```diff
@@ -0,0 +1,140 @@
...
+
+        private Mode(String text) {
+            text_ = text;
+        }
+
+        public String getText() {
+            return text_;
+        }
+
+        public static Mode get(String text) {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java:42]: We are used to place class's members in this order:
1. constants (= static or/and final fields)
2. other fields
3. public static methods
4. constructors
5. public methods
6. private methods

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java [hunk 53]
```diff
@@ -0,0 +1,140 @@
...
+                String spinnerMode = parent.getItemAtPosition(pos).toString();
+
+                ListView listView = (ListView) getActivity().findViewById(R.id.list_view);
+
+                Mode layoutMode = Mode.get(spinnerMode);
+
+                switch (layoutMode) {
+                    case LINEAR:
+                        listView.setAdapter(linearAdapter_);
+                        relativeAdapter_.notifyDataSetChanged();
```
Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Change vice-versa (Relative and Linear)

File: app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java (hunk 50 above)
Comment (MikhailSkiy) [inline app/src/main/java/com/example/admin/lab_56_57/ActivityForLab57Fragment.java]: Fixed

**Comment by oxied** [comment]
Found a bug in lab57. Scroll to several pages then return back to the top – images will disappear.

**Comment by oxied** [comment]
Looks good.
